plugins/headlamp-plugin/bin/headlamp-plugin.js

```diff
@@ -498,50 +509,89 @@ function upgrade(packageFolder, skipPackageUpdates) {
     addMissingTemplateFiles();
     addMissingConfiguration();
 
-    if (skipPackageUpdates !== true && !upgradeHeadlampPlugin(skipPackageUpdates)) {
-      return false;
+    let failed = false;
+    let reason = '';
+    if (skipPackageUpdates !== true) {
+      if (!upgradeHeadlampPlugin()) {
+        failed = true;
+        reason = 'upgrading @kinvolk/headlamp-plugin failed.';
+      }
+      if (!failed && runCmd('npm audit fix', folder)) {
+        failed = true;
+        reason = '"npm audit fix" failed.';
+      }
+      if (!failed && runCmd('npm run format', folder)) {
+        failed = true;
+        reason = '"npm run format" failed.';
+      }
+      if (!failed && runCmd('npm run lint', folder)) {
+        failed = true;
+        reason = '"npm run lint" failed.';
+      }
+      if (!failed && runCmd('npm run tsc', folder)) {
+        failed = true;
+        reason = '"npm run tsc" failed';
+      }
+    }
+    if (failed) {
+      console.error(`Failed upgrading: "${folder}". Reason: ${reason}`);
+    } else {
+      console.log(`Successfully upgraded: "${folder}".`);
     }
-
-    console.log(`Done upgrading: "${folder}".`);
     process.chdir(oldCwd);
-    return true;
+    return !failed;
   }
 
   /**
    * Upgrade each package inside the folder.
    *
-   * @param {boolean | undefined} - If true do not update packages.
-   * @returns true if all of them are upgraded successfully.
+   * @param {fs.Dirent[]} packageFolders - folders to upgrade.
+   * @param {boolean | undefined} skipPackageUpdates - If true do not update packages.
+   * @returns {boolean} - true if all of them are upgraded successfully.
    */
-  function upgradeFolderOfPackages(skipPackageUpdates) {
-    const folders = fs.readdirSync(packageFolder, { withFileTypes: true }).filter(fileName => {
-      return (
-        fileName.isDirectory() &&
-        fs.existsSync(path.join(packageFolder, fileName.name, 'package.json'))
-      );
-    });
-
-    folders.forEach(folder => {
-      const folderToBuild = path.join(packageFolder, folder.name);
-      if (!upgradePackage(folderToBuild, skipPackageUpdates)) {
-        console.error(`"${folderToBuild}" does not contain a package. Not upgrading.`);
+  function upgradeFolderOfPackages(packageFolders, skipPackageUpdates) {
+    let failed = '';
+
+    for (const folder of packageFolders) {
+      if (failed) {
```

Comment:
umm, I think we here are testing failed value for each next iteration of the loop, will this work if there were no failures till the second last packageFolders but the last one was the failed one??

Reply:
Good question. 

For the case of last one failed:

- It will return false, and shows a message about the package with an error. The log is emitted inside upgradePackage(...) when there is a failure.
- It will Not show a message about skipping any, because none will be skipped.

Reply:
aah i see yupp this makes sense, thanks